src/python/pants/jvm/resolve/coursier_setup.py

```diff
@@ -157,13 +166,26 @@ def args(self, args: Iterable[str], *, wrapper: Iterable[str] = ()) -> tuple[str
             *args,
         )
 
+    @property
+    def _coursier_cache_prefix(self) -> str:
+        """Returns a key for `COURSIER_CACHE` determined by the configured repositories.
+
+        This helps us avoid a cache poisoning issue that we uncovered in #14577.
+        """
+        sha = sha256()
+        for repo in self.repos:
+            sha.update(repo.encode("utf-8"))
+        return sha.digest().hex()
```

Comment:
I could probably store this value, but I'm not entirely certain the performance hit is great enough to justify this.

Reply:
Probably worth using `@memoized_property`

Reply:
Great! That's what I was looking for.